lib/rules/no-multi-comp.js

```diff
@@ -46,6 +51,43 @@ module.exports = {
   create: Components.detect((context, components, utils) => {
     const configuration = context.options[0] || {};
     const ignoreStateless = configuration.ignoreStateless || false;
+    const ignorePrivate = configuration.ignorePrivate || false;
+
+    const exportedComponents = new Set(); // Track exported components
+    const validIdentifiers = new Set(['ArrowFunctionExpression', 'Identifier', 'FunctionExpression']);
```

Comment:
i believe in react 19, async functions can be components as well.

Reply:
Yeah good point, I had not thought about RSCs.

Reply:
Added a little coverage for this - I'm not an RSC dev by day though so may have made some oversights.